java/ql/src/semmle/code/java/frameworks/spring/SpringUi.qll

```diff
@@ -0,0 +1,62 @@
+/**
+ * Provides models for the `org.springframework.ui` package.
+ */
+
+import java
+private import semmle.code.java.dataflow.ExternalFlow
+
+private class FlowSummaries extends SummaryModelCsv {
+  override predicate row(string row) {
+    row =
+      [
+        "org.springframework.ui;Model;true;addAllAttributes;;;Argument[-1];ReturnValue;value",
+        "org.springframework.ui;Model;true;addAllAttributes;(Collection);;Element of Argument[0];MapValue of Argument[-1];value",
+        "org.springframework.ui;Model;true;addAllAttributes;(Collection);;Element of Argument[0];MapValue of ReturnValue;value",
+        "org.springframework.ui;Model;true;addAllAttributes;(Map);;MapKey of Argument[0];MapKey of Argument[-1];value",
+        "org.springframework.ui;Model;true;addAllAttributes;(Map);;MapValue of Argument[0];MapValue of Argument[-1];value",
+        "org.springframework.ui;Model;true;addAllAttributes;(Map);;MapKey of Argument[0];MapKey of ReturnValue;value",
+        "org.springframework.ui;Model;true;addAllAttributes;(Map);;MapValue of Argument[0];MapValue of ReturnValue;value",
```

Comment:
I believe these two lines are superfluous - they'll be inferred from the two above in combination with the value step from `Argument[-1]` to `ReturnValue`. Looks like there are also a few more instances of this pattern below.